extensions/amp-script/0.1/amp-script.js

```diff
@@ -184,6 +175,105 @@ export class AmpScript extends AMP.BaseElement {
   }
 }
 
+/**
+ * A DOMPurify wrapper that implements the worker-dom.Sanitizer interface.
+ */
+class SanitizerImpl {
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @private {!DomPurifyDef} */
+    this.purifier_ = createPurifier(dict({'IN_PLACE': true}));
+
+    /** @const @private {!Element} */
+    this.wrapper_ = win.document.createElement('div');
+  }
+
+  /**
+   * @param {!Node} node
+   * @return {boolean}
+   */
+  sanitize(node) {
+    // DOMPurify sanitizes unsafe nodes by detaching them from parents.
+    // So, an unsafe `node` that has no parent will cause a runtime error.
+    // To avoid this, wrap `node` in a <div> if it has no parent.
+    const useWrapper = !node.parentNode;
+    if (useWrapper) {
+      this.wrapper_.appendChild(node);
+    }
+    const parent = node.parentNode || this.wrapper_;
+    this.purifier_.sanitize(parent);
+    const clean = parent.firstChild;
+    if (!clean) {
+      dev().info(TAG, 'Sanitized node:', node);
+      return false;
+    }
+    // Detach `node` if we used a wrapper div.
+    if (useWrapper) {
+      while (this.wrapper_.firstChild) {
+        this.wrapper_.removeChild(this.wrapper_.firstChild);
+      }
+    }
+    return true;
+  }
+
+  /**
+   * @param {!Node} node
+   * @param {string} attribute
+   * @param {string} value
+   * @return {boolean}
+   */
+  mutateAttribute(node, attribute, value) {
```

Comment:
This seems like enough code to warrant a separate test or two?

Reply:
Done.